api/v1beta1/azurecluster_default.go

```diff
@@ -59,6 +59,7 @@ func (c *AzureCluster) setNetworkSpecDefaults() {
 	c.setAPIServerLBDefaults()
 	c.SetNodeOutboundLBDefaults()
 	c.SetControlPlaneOutboundLBDefaults()
+	c.SetBackendPoolNameDefault()
```

Comment:
the unit tests for the LB default functions above are now failing because the defaulting is happening outside of the functions now... You can either 1) change the unit tests to not expect backend pool names after setting LB defaults (but then we lose unit test coverage, so you'd have to add a separate unit test), or 2) leave the `SetBackendPoolNameDefault` function in cluster.go and call `SetAPIServerLBBackendPoolNameDefault` from inside `setAPIServerLBDefaults` (and same with the other two). That way defaulting webhook code doesn't change  (it's an equivalent refactor, we're just extracting functions), and we can just use `SetBackendPoolNameDefault` in the reconciler.